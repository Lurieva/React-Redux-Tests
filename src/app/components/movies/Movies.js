import React from 'react';
import PropTypes from 'prop-types';

import Movie from './movie/Movie';
import NoFilmsFound from '../noFilmsFound/NoFilmsFound';

import './movies.scss';

const Movies = ({ movies }) => (
    <div className="movies-container">
        {
            movies.length ? movies.map((movie) => (               
                <Movie key={movie.id} movie={movie}/>
            )) : <NoFilmsFound />
        }
    </div>
);

Movies.propTypes = {
    movies: PropTypes.array
};

export default Movies;