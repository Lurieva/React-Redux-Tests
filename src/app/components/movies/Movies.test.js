import React from 'react';
import { shallow } from 'enzyme';
import Movies from './Movies';

jest.mock('./movie/Movie', () => 'Movie');
jest.mock('../noFilmsFound/NoFilmsFound', () => 'NoFilmsFound');

describe('Movies', () => {
    const movies = [{id: 1, name: 'movie_1'}, {id:2, name: 'movie_2'}];

    it('should be defined', () => {
        expect(Movies).toBeDefined();
    });

    it('should render correctly ', () => {
        const tree = shallow(<Movies movies={movies} />);
        expect(tree).toMatchSnapshot();
    });

    it('should render components as movies length', () => {
        const tree = shallow(<Movies movies={movies} />);
        expect(tree.children().length).toBe(2);
    });

    it('should render one component if there is not movies', () => {
        const tree = shallow(<Movies movies={[]} />);
        expect(tree.children().length).toBe(1);
    });
});