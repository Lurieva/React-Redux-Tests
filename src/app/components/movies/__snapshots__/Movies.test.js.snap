// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies should render correctly  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movies
    movies={
      Array [
        Object {
          "id": 1,
          "name": "movie_1",
        },
        Object {
          "id": 2,
          "name": "movie_2",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Movie
          movie={
            Object {
              "id": 1,
              "name": "movie_1",
            }
          }
        />,
        <Movie
          movie={
            Object {
              "id": 2,
              "name": "movie_2",
            }
          }
        />,
      ],
      "className": "movies-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "movie": Object {
            "id": 1,
            "name": "movie_1",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "Movie",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "movie": Object {
            "id": 2,
            "name": "movie_2",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "Movie",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Movie
            movie={
              Object {
                "id": 1,
                "name": "movie_1",
              }
            }
          />,
          <Movie
            movie={
              Object {
                "id": 2,
                "name": "movie_2",
              }
            }
          />,
        ],
        "className": "movies-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "movie": Object {
              "id": 1,
              "name": "movie_1",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "Movie",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "movie": Object {
              "id": 2,
              "name": "movie_2",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "Movie",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
